Library/ViewModels/SearchViewModel.swift

```diff
@@ -185,10 +185,21 @@ public final class SearchViewModel: SearchViewModelType, SearchViewModelInputs,
 
     self.scrollToProjectRow = self.transitionedToProjectRowAndTotalProperty.signal.skipNil().map(first)
 
-    // KSRAnalytics
+    // Tracking
+
+    let searchResults = Signal.merge(
+      self.projects,
+      viewWillAppearNotAnimated.mapConst([Project]()).take(first: 1)
+    )
 
-    viewWillAppearNotAnimated
-      .observeValues { AppEnvironment.current.ksrAnalytics.trackProjectSearchView() }
+    Signal.combineLatest(query, searchResults)
+      .takeWhen(viewWillAppearNotAnimated)
+      .observeValues { query, results in
+        let projects = query.isEmpty ? [] : results
+        let params = .defaults |> DiscoveryParams.lens.query .~ query
+        AppEnvironment.current.ksrAnalytics
+          .trackProjectSearchView(params: params, results: projects.count)
```

Comment:
Working fine except we are only returning the paginated results instead of the total. We spoke about the details offline. I think once that is resolved we are good to go here. Fixing that will potentially allow for a refactor to remove the need for searchResults as it's own constant.